linera-service/src/proxy.rs

```diff
@@ -57,8 +64,29 @@ impl Proxy {
     }
 
     /// Constructs and configures the [`Proxy`] given [`ProxyOptions`].
-    fn from_options(options: ProxyOptions) -> Result<Self> {
-        let config = ValidatorServerConfig::read(&options.config_path)?;
+    async fn from_options(options: ProxyOptions) -> Result<Self> {
+        let mut tmp_config = ValidatorServerConfig::read(&options.config_path)?;
+        if options.kube {
+            let pod_name_to_ip = fetch_pod_ips().await?;
+
+            // Assumes that each service will have just one pod. Since we have one service per shard, the first pod will always have
+            // -0 at the end of it for servers and some hash for the validators. So we check the prefix and assume only one entry
+            // will have that prefix.
+            tmp_config.validator.network.host =
+                find_value_with_key_prefix(&pod_name_to_ip, &tmp_config.validator.network.host)
+                    .expect("All hosts should be a prefix of a pod name!");
+            tmp_config.internal_network.host =
+                find_value_with_key_prefix(&pod_name_to_ip, &tmp_config.internal_network.host)
+                    .expect("All hosts should be a prefix of a pod name!");
```

Comment:
(Just for my own understanding: With the default config, the first call will just return the IP address itself without looking it up in the map, and the second one will find the pod that starts with `validator-1`? Which is the one running `linera-proxy`?)

Reply:
What do you mean exactly by "first call" and "second call"? When this is called it will lookup the pod name in the `host` field in the map to find the IP corresponding to that pod name, then replace it in the struct